watson-developer-cloud/discovery-components
# chore: add error boundary to all components

#### What do these changes do/fix?
Adds `react-error-boundary` to all components except for `DocumentPreview` as well as `CIDocument` since they were already using the custom utility `withErrorBoundary`. If we are interested in moving these components onto `react-error-boundary` for the sake of consistency I can migrate these components as well


#### How do you test/verify these changes?
to verify these changes I added `throw new Error('This is a new error');` inside of our components to test the fallback component they rendered. additionally I checked that the error and component stack were logged in the console. See example below
<img width="1538" alt="Screen Shot 2019-12-16 at 9 12 22 PM" src="https://user-images.githubusercontent.com/10775718/70958957-cac71400-2048-11ea-9f8f-e8400e2ea6dc.png">


#### Have you documented your changes (if necessary)?
N/A
#### Are there any breaking changes included in this pull request?
no
